saleor/cart/forms.py

```diff
@@ -84,6 +84,7 @@ class ReplaceCartLineForm(AddToCartForm):
     def __init__(self, *args, **kwargs):
         super(ReplaceCartLineForm, self).__init__(*args, **kwargs)
         self.cart_line = self.cart.get_line(self.product)
+        self.fields['quantity'].widget.attrs = {'min': 1, 'max': self.cart_line.product.get_stock_quantity()}
```

Comment:
The cart is not guaranteed to contain the product. We can't rely on `self.cart_line` not being `None` here. `self.product.get_stock_quantity()` should work though.